packages/lisk-transactions/src/base_transaction.ts

```diff
@@ -0,0 +1,497 @@
+		};
+
+		if (!this.signature) {
+			return transaction;
+		}
+
+		const singleSignedTransaction = {
+			...transaction,
+			signature: this.signature,
+		};
+
+		if (!this.signSignature) {
+			return singleSignedTransaction;
+		}
+
+		const signedTransaction = {
+			...singleSignedTransaction,
+			signSignature: this.signSignature,
+		};
+
+		return signedTransaction;
+	}
+
+	protected getBasicBytes(): Buffer {
+		const transactionType = Buffer.alloc(BYTESIZES.TYPE, this.type);
+		const transactionTimestamp = Buffer.alloc(BYTESIZES.TIMESTAMP);
+		transactionTimestamp.writeIntLE(this.timestamp, 0, BYTESIZES.TIMESTAMP);
+
+		const transactionSenderPublicKey = cryptography.hexToBuffer(
+			this.senderPublicKey,
+		);
+
+		const transactionRecipientID = this.recipientId
+			? cryptography.bigNumberToBuffer(
+					this.recipientId.slice(0, -1),
+					BYTESIZES.RECIPIENT_ID,
+			  )
+			: Buffer.alloc(BYTESIZES.RECIPIENT_ID);
+
+		const transactionAmount = this.amount.toBuffer({
+			endian: 'little',
+			size: BYTESIZES.AMOUNT,
+		});
+
+		return Buffer.concat([
+			transactionType,
+			transactionTimestamp,
+			transactionSenderPublicKey,
+			transactionRecipientID,
+			transactionAmount,
+		]);
+	}
+
+	public abstract getBytes(): Buffer;
+
+	public abstract containsUniqueData(): boolean;
+
+	public checkSchema(): TransactionResponse {
+		const transaction = this.toJSON();
+		const baseTransactionValidator = validator.compile(schemas.baseTransaction);
+		const valid = baseTransactionValidator(transaction) as boolean;
+
+		const errors = baseTransactionValidator.errors
+			? baseTransactionValidator.errors.map(
+					error =>
+						new TransactionError(
+							`'${error.dataPath}' ${error.message}`,
+							transaction.id,
+							error.dataPath,
+						),
+			  )
+			: [];
+
+		return {
+			id: this.id,
+			status: valid ? Status.OK : Status.FAIL,
+			errors,
+		};
+	}
+
+	public validate(): TransactionResponse {
+		const transaction = this.toJSON();
+		const transactionHash = cryptography.hash(this.getBasicBytes());
+
+		const validateErrors = Object.entries(transaction).reduce(
+			(
+				errorArray: ReadonlyArray<TransactionError>,
+				property: ReadonlyArray<string & ReadonlyArray<string>>,
+			): ReadonlyArray<TransactionError> => {
+				const [key, value] = property;
+				if (key === 'id') {
+					const transactionBytesWithSignatures = getTransactionBytes(
+						transaction,
+					);
+					const transactionHashWithSignatures = cryptography.hash(
+						transactionBytesWithSignatures,
+					);
+					const bufferFromFirstEntriesReversed = cryptography.getFirstEightBytesReversed(
+						transactionHashWithSignatures,
+					);
+					const transactionId = cryptography.bufferToBigNumberString(
+						bufferFromFirstEntriesReversed,
+					);
+					if (value !== transactionId) {
+						return [
+							...errorArray,
+							new TransactionError('Invalid transaction id', this.id, '.id'),
+						];
+					}
+				}
+				if (key === 'senderId') {
```

Comment:
this could be part of `checkShema`?
I think conclusion was
constructor => basic type check
checkSchema => detailed field check (formatting etc) and cross field check (senderID = senderPublicKey)
validation => signature check
verification => state check